src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2152,6 +2153,17 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
 		HashSet<String> instsWithSeparator = new HashSet<>();
 		for(Entry<String,String> e : srcToDestInstNames.entrySet()) {
 			EDIFHierCellInst cellInst = src.getNetlist().getHierCellInstFromName(e.getKey());
+			if(e.getValue().length() == 0) {
+			    // If its the top cell, remove the top cell from destNetlist
```

Comment:
Similar to my other comment, is this a destructive operation on the source netlist?

Reply:
Also, it looks like this method was changed but I'm not seeing the linkage with the rest of this PR?

Reply:
`copyImplementation()` is a destructive operation on the source design, although maybe it shouldn't be.  These changes aren't technically related to the ones below--but they were necessary in trying to solve the same issue.  Should I change the PR title or create a new PR?

Reply:
I would prefer to put it into a different PR to keep this one clean.

Reply:
See #395

Reply:
Looks like this one includes some of the changes in #395, but #395 also depends on this PR?
Does the whole of `DesignTools.java` need to be reverted in this PR and for this to be merged in first?

Reply:
I plan to merge in this order, bringing in the latest to each PR after each merge: #396, #385, and #395.

Reply:
Roger that. Too many numbers!